js_modules/dagster-ui/packages/ui-core/src/pipelines/PipelineExplorerRoot.tsx

```diff
@@ -80,12 +81,24 @@ export const PipelineExplorerContainer = ({
   const parentNames = explorerPath.opNames.slice(0, explorerPath.opNames.length - 1);
   const pipelineSelector = buildPipelineSelector(repoAddress || null, explorerPath.pipelineName);
 
+  const snapIdFromContext = useJob(repoAddress || null, explorerPath.pipelineName)
+    ?.pipelineSnapshotId;
+  const snapshotId = useMemo(() => {
+    return snapIdFromContext
+      ? snapIdFromContext
+      : explorerPath.snapshotId
+      ? explorerPath.snapshotId
+      : undefined;
+    // only add snapshot id from workspace if its loaded when we first render
```

Comment:
// only add snapshot id from workspace if its loaded when we first render in order to avoid changing query variables which would trigger another fetch

?